src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,48 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, V> {
+    private T first;
+    private V second;
 
+    private Pair(T first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public V getSecond() {
+        return second;
+    }
+
+    public static <T, V> Pair<T, V> of(T first, V second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        Pair<T, V> pair = (Pair<T, V>) object;
+        return ((first == pair.first) || first != null && first.equals(pair.first)
+                && (second == pair.second) || (second != null && second.equals(pair.second)));
```

Comment:
at the moment this part `first != null && first.equals(pair.first) && (second == pair.second)` is evaluated together

Reply:
done